Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Material/MaterialAsset.h

```diff
@@ -105,6 +105,7 @@ namespace AZ
             //! their parent material, they all get flattened at build time so every MaterialAsset has the full set of values.
             AZStd::array_view<MaterialPropertyValue> GetPropertyValues() const;
 
+            const AZStd::vector<AZ::Name>& GetPropertyNames() const;
```

Comment:
Needs an API comment. This should explain that the property names list might be empty, and what the implications are.

Also, the comment for GetPropertyValues needs to be updated. The part about "entries in this list align" is true when there are no names available, but when they are available, then GetPropertyNames must be used to align the indices.

Reply:
Actually, I don't really like having APIs that have different operating modes like this, and I'm trying to think how we could improve this. One idea is to add a GetAlignedPropertyValues() function that populates a vector with property values by looking up each name and putting them in the right index (unless there are no property names in which case it just returns m_propertyValues. There are other approaches we could take; maybe spend some time thinking about this, and see if you can come up with a more consistent API ;)

Reply:
Can we just enforce the alignment of m_propertyValues by pre-processing with m_propertyNames if it isn't empty? That way we don't need to touch the API.